pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/notPepperKit/NotPepperKitCreationService.java

```diff
@@ -0,0 +1,238 @@
+    public final static String UNKNOWN_KIT_TYPE = "UNKNOWN_KIT_TYPE";
+    public final static String UNKNOWN_STUDY = "UNKNOWN_STUDY";
+    public final static String MISSING_JUNIPER_KIT_ID ="MISSING_JUNIPER_KIT_ID";
+    public final static String MISSING_JUNIPER_PARTICIPANT_ID ="MISSING_JUNIPER_PARTICIPANT_ID";
+    public final static String JUNIPER ="JUNIPER";
+    public final static String JUNIPER_UNDERSCORE ="JUNIPER_";
+
+    public KitResponse createNonPepperKit(JuniperKitRequest juniperKitRequest, String studyGuid, String kitTypeName) {
+        if (StringUtils.isBlank(juniperKitRequest.getJuniperParticipantID())) {
+            return new KitResponse(MISSING_JUNIPER_PARTICIPANT_ID, juniperKitRequest.getJuniperKitId());
+        }
+        if (StringUtils.isBlank(juniperKitRequest.getJuniperKitId())) {
+            return new KitResponse(MISSING_JUNIPER_KIT_ID, null);
+        }
+        Optional<DDPInstanceDto> maybeDdpInstanceDto = new DDPInstanceDao().getDDPInstanceByStudyGuid(studyGuid);
+        if (maybeDdpInstanceDto.isEmpty()) {
+            return new KitResponse(UNKNOWN_STUDY, juniperKitRequest.getJuniperKitId());
+        }
+        DDPInstanceDto ddpInstanceDto = maybeDdpInstanceDto.get();
+        //getting the instance with isHasRole being set to true if the instance has role juniper_study
+        DDPInstance ddpInstance = DDPInstance.getDDPInstanceWithRole(ddpInstanceDto.getInstanceName(), "juniper_study");
+
+        if (!ddpInstance.isHasRole()) {
+            throw new RuntimeException("This is not a Juniper study!");
+        }
+        HashMap<String, KitType> kitTypes = KitType.getKitLookup();
+        String key = kitTypeName + "_" + ddpInstanceDto.getDdpInstanceId();
+        KitType kitType = kitTypes.get(key);
+        if (kitType == null) {
+            return new KitResponse(UNKNOWN_KIT_TYPE, juniperKitRequest.getJuniperKitId());
+        }
+
+        Map<Integer, KitRequestSettings> kitRequestSettingsMap =
+                KitRequestSettings.getKitRequestSettings(String.valueOf(ddpInstanceDto.getDdpInstanceId()));
+        KitRequestSettings kitRequestSettings = kitRequestSettingsMap.get(kitType.getKitTypeId());
+
+        // if the kit type has sub kits > like for testBoston
+//        boolean kitHasSubKits = kitRequestSettings.getHasSubKits() != 0;
+
+        log.info("Setup EasyPost...");
+        EasyPostUtil easyPostUtil = new EasyPostUtil(ddpInstanceDto.getInstanceName());
+
+        if (!checkAddress(juniperKitRequest, kitRequestSettings.getPhone(), easyPostUtil)) {
+            return new KitResponse(ADDRESS_VALIDATION_ERROR, juniperKitRequest.getJuniperKitId());
+        }
+
+        ArrayList<KitRequest> orderKits = new ArrayList<>();
+
+
+        TransactionWrapper.inTransaction(conn -> {
+            createKit(ddpInstance, kitType, juniperKitRequest, kitRequestSettings, easyPostUtil, kitTypeName,  orderKits, conn);
```

Comment:
Not sure what's going on here. How do we know if the kit was created successfully?